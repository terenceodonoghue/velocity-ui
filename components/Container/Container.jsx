/** @jsx jsx */
import { jsx } from '@emotion/core';
import PropTypes from 'prop-types';
import * as S from './Container.styles';

const Card = ({ children, title, ...rest }) => (
  <S.Card {...rest}>
    {title && <S.Title>{title}</S.Title>}
    {children}
  </S.Card>
);

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string,
};

Card.defaultProps = {
  title: null,
};

export default {
  Card,
};
