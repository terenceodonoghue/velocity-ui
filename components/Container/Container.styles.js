import styled from '@emotion/styled';

export const Card = styled.div`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  background-color: var(--color-white);
`;

export const Heading = styled.h3`
  margin: 0 0 26px 0;
  color: var(--color-bluey-grey);
  font-size: 13px;
  font-weight: var(--font-weight-normal);
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
