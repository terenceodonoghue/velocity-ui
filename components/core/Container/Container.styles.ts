import { css } from '@emotion/react';

export const container = css`
  position: relative;
  max-width: 1360px;
  padding: 64px 110px;
`;

export const heading = css`
  margin: 0 0 24px;
`;
