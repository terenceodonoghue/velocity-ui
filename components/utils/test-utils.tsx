import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
