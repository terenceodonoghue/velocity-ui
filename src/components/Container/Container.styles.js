import styled from '@emotion/styled';

export const Card = styled.div`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  padding: 32px 24px;
  background-color: ${(props) => props.theme.colors.white};
`;

export const Heading = styled.h3`
  margin: 0 0 26px 0;
  color: ${(props) => props.theme.colors.blueyGrey};
  font-size: 13px;
  font-weight: ${(props) => props.theme.typography.fontWeightRegular};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
