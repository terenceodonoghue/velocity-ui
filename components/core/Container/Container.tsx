import React, { FunctionComponent } from 'react';
import * as css from './Container.styles';
import type { ContainerProps } from './Container.types';

const Container: FunctionComponent<ContainerProps> = ({
  children,
  heading,
  ...props
}) => (
  <>
    {heading && (
      <h1 css={css.heading} data-testid="container-heading">
        {heading}
      </h1>
    )}
    <div css={css.container} data-testid="container" {...props}>
      {children}
    </div>
  </>
);

export default Container;
