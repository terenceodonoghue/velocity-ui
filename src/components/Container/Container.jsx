/** @jsx jsx */
import { jsx } from '@emotion/core';
import React from 'react';
import PropTypes from 'prop-types';
import * as S from './Container.styles';

const Card = React.forwardRef(({ children, heading, ...rest }, ref) => (
  <S.Card ref={ref} {...rest}>
    {heading && <S.Heading>{heading}</S.Heading>}
    {children}
  </S.Card>
));

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  heading: PropTypes.string,
};

Card.defaultProps = {
  heading: null,
};


export default {
  Card,
};
