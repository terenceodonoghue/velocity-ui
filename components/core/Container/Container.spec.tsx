import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import Container from './Container';

describe('Container', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId, queryByTestId } = render(
      <Container>{content}</Container>,
    );

    expect(getByTestId('container')).toHaveTextContent(content);
    expect(queryByTestId('container-heading')).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders content with heading', () => {
    const content = faker.hacker.phrase();
    const heading = faker.hacker.phrase();

    const { container, getByTestId } = render(
      <Container heading={heading}>{content}</Container>,
    );

    expect(getByTestId('container')).toHaveTextContent(content);
    expect(getByTestId('container-heading')).toHaveTextContent(heading);
    expect(container).toMatchSnapshot();
  });
});
