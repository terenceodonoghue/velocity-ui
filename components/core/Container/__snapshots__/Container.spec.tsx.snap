// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders content 1`] = `
.emotion-0 {
  position: relative;
  max-width: 1360px;
  padding: 64px 110px;
}

<div>
  <div
    class="emotion-0"
    data-testid="container"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </div>
</div>
`;

exports[`Container renders content with heading 1`] = `
.emotion-0 {
  margin: 0 0 24px;
}

.emotion-1 {
  position: relative;
  max-width: 1360px;
  padding: 64px 110px;
}

<div>
  <h1
    class="emotion-0"
    data-testid="container-heading"
  >
    You can't input the array without navigating the redundant GB alarm!
  </h1>
  <div
    class="emotion-1"
    data-testid="container"
  >
    You can't quantify the interface without parsing the neural ADP feed!
  </div>
</div>
`;
